src/status_im2/subs/shell.cljs

```diff
@@ -147,3 +147,35 @@
    (let [community-id (:community-id channel)
          community    (get communities (:community-id channel))]
      (community-channel-card community community-id channel channel-id))))
+
+(re-frame/reg-sub
+ :shell/bottom-tabs-notifications-data
+ :<- [:chats/chats]
+ (fn [chats]
+   (let [chats-seq                      (map second chats)
+         unviewed-messages-count-seq    (map #(hash-map (:chat-type %)
```

Comment:
I think this can be done in one go with a reduce, it should be faster and a bit simpler, if I understand correctly, you want to get a counter for mentions/messages, something like that should work (pseudo-code/ish and you can extract and test separately, also you can make other decisions in terms of how you want the accumulator to look like, if you want already to group private/one to one in the reduce etc):

```
(reduce
   (fn [acc {:keys [unviewed-messages-count unviewed-mentions-count chat-type]}]
       (if (chat-type is not one we care about)
          acc
          (-> acc
             (update-in [chat-type :mentions-count] + unviewed-mentions-count) // adding nil is ok, it's counted as 0, you can coherce if you want though)
             (update-in [chat-type :unviewed-mentions-count] + unviewed-mentions-count))
            
)
    {community-chat-type {:mentions-count 0 :messages-count}
      one-to-one-chat-type {:mentions-count 0 :messages-count}}
     chats)
  ```

Reply:
I actually was implementing this with reduce initially. But don't know why that looked more complex (maybe because of too many ifs).

But yes with reduce time taken is decreased from 0.91 ms to 0.46 ms :tada:. Thank you very much.

Reply:
looking at second suggestion

Reply:
tried filtering input of reduce function using

```
          (filter #(#{constants/one-to-one-chat-type
                      constants/community-chat-type
                      constants/private-group-chat-type}
                    (:chat-type %)) (vals chats)))
```
But that increased time to `~0.65`